src/network/peer.zig

```diff
@@ -20,11 +20,45 @@ pub const Peer = struct {
 
     /// Initialize a new peer
     pub fn init(allocator: std.mem.Allocator, config: *const Config, address: std.net.Address) !*Peer {
-        if (address.any.family != std.posix.AF.INET6) {
-            return error.WeOnlySupportIPV6ForNow;
+        const Family = std.posix.AF;
+        var final_address: std.net.Address = address;
+        var stream: std.net.Stream = undefined;
+
+        // Check if the address is IPv4 or IPv6
+        if (address.any.family != Family.INET6 and address.any.family != Family.INET) {
+            return error.UnsupportedAddressFamily;
+        }
+
+        if (address.any.family == Family.INET) {
+            // Convert IPv4 to IPv6-mapped IPv4 address
+            const ipv4_address = address.in;
+
+            var ipv6_mapped: [16]u8 = [_]u8{0} ** 16;
+
+            // Set bytes 10 and 11 to 0xff
+            ipv6_mapped[10] = 0xff;
+            ipv6_mapped[11] = 0xff;
+
+            // Copy the IPv4 address into the last 4 bytes of the IPv6-mapped address
+            const ipv4_bytes = std.mem.asBytes(&ipv4_address.sa.addr);
+            ipv6_mapped[12] = ipv4_bytes[0];
+            ipv6_mapped[13] = ipv4_bytes[1];
+            ipv6_mapped[14] = ipv4_bytes[2];
+            ipv6_mapped[15] = ipv4_bytes[3];
+
+            final_address = std.net.Address{
+                .in6 = .{
+                    .sa = std.posix.sockaddr.in6{
+                        .addr = ipv6_mapped,
+                        .port = ipv4_address.getPort(),
+                        .flowinfo = 0,
+                        .scope_id = 0,
+                    },
+                },
+            };
         }
 
-        const stream = try std.net.tcpConnectToAddress(address);
+        stream = try std.net.tcpConnectToAddress(final_address);
```

Comment:
We can keep using the original address there.

We need the conversion for when we are doing the sendVersionMessage thing